src/pages/settings/Profile/PersonalDetails/AddressPage.js

```diff
@@ -52,175 +52,153 @@ const defaultProps = {
-                            maxLength={CONST.FORM_CHARACTER_LIMIT}
-                        />
-                    </View>
-                    <View style={styles.mb4}>
-                        <TextInput
-                            inputID="city"
-                            label={this.props.translate('common.city')}
-                            defaultValue={address.city || ''}
-                            maxLength={CONST.FORM_CHARACTER_LIMIT}
-                        />
-                    </View>
-                    <View style={styles.mb4}>
-                        {this.state.isUsaForm ? (
-                            <StatePicker
-                                inputID="state"
-                                defaultValue={address.state || ''}
-                            />
-                        ) : (
-                            <TextInput
-                                inputID="state"
-                                label={this.props.translate('common.stateOrProvince')}
-                                defaultValue={address.state || ''}
-                                maxLength={CONST.FORM_CHARACTER_LIMIT}
-                            />
-                        )}
-                    </View>
-                    <View style={styles.mb4}>
-                        <TextInput
-                            inputID="zipPostCode"
-                            label={this.props.translate('common.zipPostCode')}
-                            autoCapitalize="characters"
-                            defaultValue={address.zip || ''}
-                            maxLength={CONST.BANK_ACCOUNT.MAX_LENGTH.ZIP_CODE}
-                            hint={this.state.zipFormat}
-                        />
-                    </View>
-                    <View>
-                        <CountryPicker
-                            inputID="country"
-                            onValueChange={this.onCountryUpdate}
-                            defaultValue={PersonalDetails.getCountryISO(address.country)}
-                        />
-                    </View>
-                </Form>
-            </ScreenWrapper>
-        );
-    }
+    return errors;
+}
+
+/**
+ * Submit form to update user's first and last legal name
+ * @param {Object} values - form input values
+ */
+const updateAddress = (values) => {
+    PersonalDetails.updateAddress(values.addressLine1.trim(), values.addressLine2.trim(), values.city.trim(), values.state.trim(), values.zipPostCode.trim(), values.country);
+};
+
+const AddressPage = (props) => {
+    const [countryISO, setCountryISO] = useState(PersonalDetails.getCountryISO(lodashGet(props.privatePersonalDetails, 'address.country')) || CONST.COUNTRY.US);
+    const isUsaForm = countryISO === CONST.COUNTRY.US;
+    
+    const zipSampleFormat = lodashGet(CONST.COUNTRY_ZIP_REGEX_DATA, [countryISO, 'samples'], '');
+    const zipFormat = props.translate('common.zipCodeExampleFormat', {zipSampleFormat});
```

Comment:
The only real state is `countryISO` (or country), `zipFormat` is derived from `countryISO`